samples/fragment-payment-processor-settlement.js

```diff
@@ -0,0 +1,93 @@
+function() {
+  // some common reuseable stuff
+  var docBusinessId = getBusinessId(doc, oldDoc);
+  var PROCESSOR_ID_MATCH_GROUP = 1;
+  var SETTLEMENT_ID_MATCH_GROUP = 2;
+  var typeRegex = createBusinessEntityRegex('paymentProcessor\\.([A-Za-z0-9_-]+)\\.processedSettlement\\.([A-Za-z0-9_-]+)');
+  var typeRegexMatchGroups = typeRegex.exec(doc._id);
+
+  return {
+    typeFilter: function(doc, oldDoc) {
+      return typeRegex.test(doc._id);
+    },
+    channels: function(doc, oldDoc) {
+      // Only staff/service users can create, replace or delete payment processor settlements to prevent regular users from tampering
+      return {
+        view: toSyncChannel(docBusinessId, 'VIEW_PAYMENT_PROCESSOR_SETTLEMENTS'),
+        write: 'payment-settlement-write'
+      };
+    },
+    authorizedRoles: defaultAuthorizedRoles,
+    authorizedUsers: defaultAuthorizedUsers,
+    immutable: true,
+    propertyValidators: {
+      businessId: {
+        // The ID of the business with which the settlement is associated
+        type: 'integer',
+        required: true,
+        minimumValue: Number(docBusinessId),
+        maximumValue: Number(docBusinessId),
+        immutable: true
+      },
+      transferId: {
+        // The ID of the Books transfer record that represents the settlement
+        type: 'integer',
+        required: true,
+        minimumValue: 1,
+        immutable: true
+      },
+      settlementId: {
+        // The ID of the settlement as provided by the payment processor
+        type: 'string',
+        required: true,
+        mustNotBeEmpty: true,
+        immutable: true,
+        regexPattern: function(doc, oldDoc, value, oldValue) {
+          var expectedSettlementId = typeRegexMatchGroups[SETTLEMENT_ID_MATCH_GROUP];
+
+          return new RegExp('^' + expectedSettlementId + '$');
+        }
+      },
+      processorId: {
+        // The Kashoo payment processor associated with the settlement
+        type: 'string',
+        required: true,
+        mustNotBeEmpty: true,
+        immutable: true,
+        regexPattern: function(doc, oldDoc, value, oldValue) {
+          var expectedSettlementId = typeRegexMatchGroups[PROCESSOR_ID_MATCH_GROUP];
```

Comment:
nitpick:  variable name doesn't jive with field name - likely an artifact of c/p from the field above.